rally-domain/src/main/groovy/yakworks/rally/orgs/repo/ContactRepo.groovy

```diff
@@ -82,21 +87,32 @@ class ContactRepo extends LongIdGormRepo<Contact> {
     @Override
     void doBeforePersistWithData(Contact contact, PersistArgs args) {
         Map data = args.data
+        if (args.bindAction == BindAction.Create) {
+            setupSource(contact, data)
+        }
         // we do primary location and contact here before persist so we persist org only once with contactId it is created
         if(data.location) createOrUpdateLocation(contact, data.location as Map)
     }
 
+    void setupSource(Contact contact, Map data) {
+        if(!contact.id) contact.id = generateId()
+        contact.source = ContactSource.repo.createSource(contact, data)
```

Comment:
where do you persist? look at the logic in AbstractOrgRepo

Reply:
contact source is created by source repo, and contact.source referece gets persisted when contact gets persisted.
same as org

Reply:
@snimavat I am pointing these things out to fix them, not dismiss them. persist is NOT called on the ContactSource, there should never be a time we let hibernate persist without it being explicitly called, EVER. we dont do it anywhere in our code and never should be doing it. as I said, follow the logic in the AbstractOrgRepo, doBeforePersistWithData calls the verifyNumAndOrgSource which is what sets it up there, and then it persists it. here you call setupSource and it MUST be persisted, its its no it will never hit the repo logic. not only that, but like the ArTranSource, this should/will have a foreign key tied to it, which means it has to be be persisted first. keep it consistent.

Reply:
@basejump createSource setsup source from data and does - `ContactSource.create(sourceData)` - Am i missing some thing here

Reply:
@basejump ContactSourceRepo setsup source from data and does `ontactSource.create(sourceData)` it gets persisted. It gets persisted before contact